File: VB/MyDateEdit/CalendarHitInfoDescendant.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraEditors.Calendar
Imports System.Drawing

Namespace WindowsFormsApplication2
	Public Class MyCalendarHitInfo
		Inherits CalendarHitInfo
		Public Sub New(ByVal pt As Point)
			MyBase.New(pt)

		End Sub

		' Fields...
		Private _InRightYearButton As Boolean = False
		Private _InLeftYearButton As Boolean = False

		Public Property InLeftYearButton() As Boolean
			Get
				Return _InLeftYearButton
			End Get
			Set(ByVal value As Boolean)
				_InLeftYearButton = value
			End Set
		End Property


		Public Property InRightYearButton() As Boolean
			Get
				Return _InRightYearButton
			End Get
			Set(ByVal value As Boolean)
				_InRightYearButton = value
			End Set
		End Property

		Public Sub Assign(ByVal hitInfo As CalendarHitInfo)
			Bounds = hitInfo.Bounds
			HitDate = hitInfo.HitDate
			HitObject = hitInfo.HitObject
			InfoType = hitInfo.InfoType
		End Sub

	End Class
End Namespace

File: CS/MyDateEdit/CalendarHitInfoDescendant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraEditors.Calendar;
using System.Drawing;

namespace WindowsFormsApplication2
{
    public class MyCalendarHitInfo : CalendarHitInfo
    {
        public MyCalendarHitInfo(Point pt)
            : base(pt)
        {

        }

        // Fields...
        private bool _InRightYearButton = false;
        private bool _InLeftYearButton = false;

        public bool InLeftYearButton
        {
            get { return _InLeftYearButton; }
            set { _InLeftYearButton = value; }
        }


        public bool InRightYearButton
        {
            get { return _InRightYearButton; }
            set
            {
                _InRightYearButton = value;
            }
        }

        public void Assign(CalendarHitInfo hitInfo)
        {
            Bounds = hitInfo.Bounds;
            HitDate = hitInfo.HitDate;
            HitObject = hitInfo.HitObject;
            InfoType = hitInfo.InfoType;
        }
        
    }
}
